source/codegen/templates/proto_helpers.mako

```diff
@@ -53,6 +53,48 @@ enum ${enum_name} {
 % endfor
 </%def>
 
+## Define enums in the proto per attribute type that uses enum values
+<%def name="define_attribute_values_enums(attribute_enums)">\
+<%
+  config = data["config"]
+  service_class_prefix = config["service_class_prefix"]
+  enums = data["enums"]
+%>\
+% for type_name in attribute_enums:
+<%
+  type_enum_name = service_class_prefix + type_name[2:] + "AttributeValues"
+  enum_value_type_specific_prefix = (f"{service_class_prefix.upper()}_{type_name[2:]}_VAL").upper()
+  values_to_create = { enum_value_type_specific_prefix + "_UNSPECIFIED": 0 }
```

Comment:
note: Unlike function enums, we need to create dictionary of all enum names/values before writing to output file so as to check if to set `allow_alias` by checking duplicate values across values from all the enums that we are combining.